bigquery_etl/shredder/bug1751979.py

```diff
@@ -249,3 +249,124 @@
+        CONCAT(engine, '.', source) AS key,
+        `count`
+      FROM
+        UNNEST(input)
+    ),
+    parsed AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, "([^.]+[.]in-content[:.][^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^.]+[.]in-content[:.][^:]+:(.*)") AS code,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+        * REPLACE (
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
+            ),
+            key,
+            CONCAT(prefix, "other.scrubbed")
+          ) AS key
+        )
+      FROM
+        parsed
+    )
+    -- The SUBSTR and UNNEST logic here is copied from the prod main_summary query
+    -- see https://github.com/mozilla/bigquery-etl/blob/222c4266/sql
+    --     /moz-fx-data-shared-prod/telemetry_derived/main_summary_v4/part1.sql#L266-L273
+    SELECT AS STRUCT
+      SUBSTR(_key, 0, pos - 2) AS engine,
+      SUBSTR(_key, pos) AS source,
+      -- We sum by engine and source just in case we have several non-conforming codes that
+      -- all get the same scrubbed value; we want to avoid having duplicate keys.
+      SUM(`count`) AS `count`
+    FROM
+      parsed,
+      UNNEST([REPLACE(key, 'in-content.', 'in-content:')]) AS _key,
+      UNNEST([LENGTH(REGEXP_EXTRACT(_key, '.+[.].'))]) AS pos
```

Comment:
Won't the dot in `other.scrubbed` cause problems with this logic?

It seems like the `.+` will be greedy and you'll end up with the engine ending in `in-content:other` and the source being `scrubbed`.

Reply:
I think you're right. The existing content already has this problem where the `engine` often has more than it should due to a dot in the latter part of the string.

This same logic is used in the clients_daily query. I will check clients_daily to see if the `other.scrubbed` values coming through the pipeline are already being parsed this way.

Reply:
Indeed, `clients_daily` includes entries like with engine like `bing.in-content:organic:other` and source `scrubbed`.

We need to update this logic in the source queries as well as here.

Thank you so much for scrutinizing this!

Reply:
IMO using `other-scrubbed` (like was done for mobile) would be ideal, but I suppose that ship has sailed since changing now would require re-reprocessing `main_v4`.

Reply:
Without boiling the ocean here, I think we can achieve what we want by changing `+` to `+?` which according to the [re2 docs](https://github.com/google/re2/wiki/Syntax) means "match one or more, prefer fewer".

Reply:
Sounds good, and I'm glad to see that non-greedy matching is supported in BigQuery! (coming from Snowflake, where it silently allowed non-greedy indicators but kept being greedy because it didn't really support that option)